sites/public/pages/applications/household/programs.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { useState, useEffect } from "react"
+import { useForm } from "react-hook-form"
+import {
+  AlertBox,
+  Form,
+  FormCard,
+  ProgressNav,
+  FieldGroup,
+  t,
+  Button,
+  AppearanceStyleType,
+  OnClientSide,
+} from "@bloom-housing/ui-components"
+import FormsLayout from "../../../layouts/forms"
+import FormBackLink from "../../../src/forms/applications/FormBackLink"
+import { useFormConductor } from "../../../lib/hooks"
+import {
+  mapProgramsToApi,
+  getProgramOptionName,
+  getProgramOptionDescription,
+} from "@bloom-housing/shared-helpers"
+
+const ApplicationPrograms = () => {
+  const clientLoaded = OnClientSide()
+  const { conductor, application, listing } = useFormConductor("programs")
+  const programs = listing?.listingPrograms
+  const uniquePages = new Set(programs?.map((item) => item.ordinal)).size
+  const [page, setPage] = useState(conductor.navigatedThroughBack ? uniquePages : 1)
+  const [pageProgram, setPageProgram] = useState(null)
+  const [programData, setProgramData] = useState(null)
+
+  const currentPageSection = 2
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  const { register, handleSubmit, errors, reset } = useForm({
+    defaultValues: {
+      application: { programs: {} },
+    },
+  })
+
+  /*
+    Required to keep the form up to date before submitting this section if you're moving between pages
+  */
+  useEffect(() => {
+    const findProgram = programs?.find((item) => {
+      return item.ordinal === page
+    })?.program
+    if (findProgram) {
+      setPageProgram(findProgram)
+      setProgramData(application.programs.find((item) => item.key === findProgram.formMetadata.key))
+    }
+    reset({
+      application: { programs: {} },
+    })
+  }, [page, programs, application.programs, reset])
+
+  /*
+    Submits the form
+  */
+  const onSubmit = (data) => {
+    let programsData = mapProgramsToApi(
+      programs.map((item) => item.program),
+      data
+    )
+
+    if (uniquePages > 1) {
+      // If we've split programs across multiple pages, save the data in segments
+      const programsDataKeys = programsData.map((program) => program.key)
+      const currentPrograms = application.programs.filter((program) => {
+        return !programsDataKeys.includes(program.key)
+      })
+      programsData = [...currentPrograms, ...programsData]
+      conductor.currentStep.save(programsData)
+    } else {
+      // Otherwise, submit all at once
+      conductor.currentStep.save(programsData)
+    }
```

Comment:
I think the comment on line 67 doesn't really communicate what this is. We always have to divide the total number of programs into different "pages" here since it's simulating Next's page-based navigation but within a single file. Each program ordinal (1, 2, 3) will then place that program on each of the "pages".